src/index.mjs

```diff
@@ -175,3 +222,45 @@ export function prefetch(url, isPriority, conn) {
       })
   );
 }
+
+/**
+* Prerender a given URL
+* @param {String} url - the URL to fetch
+* @param {Object} [conn] - navigator.connection (internal)
+* @return {Object} a Promise
+*/
+export function prerender(urls, conn){
+  checkConnection(conn = navigator.connection).catch(err => {
+    return Promise.reject(new Error('Cannot prerender, '+err.message));
+  });
+  
+  // prerendering preconditions:
+  // 1) whether UA supports spec rules.. If not, fallback to prefetch
+  if (!hasSpecRulesSupport()) {
+    prefetch (urls);
+    return Promise.reject(new Error('This Browser does not support Speculation Rules API. Falling back to Prefetch.'));
+  }
+  
+  // 2) whether spec rules is already defined (and with this we also covered when we have created spec rules before)
+  if (isSpecRulesExists()) {
+    return Promise.reject(new Error('Speculation Rules is already defined and cannot be altered.'));
+  }
+    
+  // 3) whether it's a same origin url,
+  try{
```

Comment:
Similar style nit to the previous comment.

Reply:
Also fixed with baeb7435c574fd21ef4ee28e42c65da13dee470a